final_task/rss_reader/functions/print_func.py

```diff
@@ -0,0 +1,72 @@
+"""Contain formatting output functions
+
+Functions
+---------
+check_limit_argument(command_line_args, news_collection, logger) -> news_collection
+    Check presence and validness of limit argument
+    If present and valid, then limit collection of news
+    Uses inside print_feeds() function
+--------------------------------------------
+generate_news_json(news_collection, logger) -> news_json
+    Create json of news
+--------------------------------------------
+print_feeds(news_collection, command_line_args,logger) -> None
+    Print news to stdout in json or text format
+"""
+
+import json
+
+
+def check_limit_argument(command_line_args, news_collection, logger):
+    """Check presence and validness of limit argument
+
+    If present and valid, then limit collection of news
+    """
+
+    if command_line_args.limit:
+        limit = command_line_args.limit
+        if len(news_collection) > limit >= 0:
+            return news_collection[:limit]
+        elif limit < 0:
+            logger.warning("Limit argument can't be a negative")
+            limit = int(input("Please, enter the valid value: "))
+            command_line_args.limit = limit
```

Comment:
It is better to have this check before everything else in this function. Moreover, if user entered wrong limit it is his fault, so just raise an exception here and stop the script.

All user inputs should be always validated and having `input()` anywhere is unsafe.

Reply:
Fixed